docs/source/features/run_workflow_remotely.md

```diff
@@ -0,0 +1,99 @@
+
+This documentation demonstrates how to configure the remote dispatcher, run the workflow, and retrieve the results of the remote workflow.
+
+## Install Extra Dependencies
+
+You can install the necessary dependencies by running:
+
+```shell
+pip install olive-ai[dispatcher]
+```
+
+or
+
+```shell
+pip install paramiko
+```
+
+## Configure a Remote Dispatcher
+
+The Remote Dispatcher is a type of dispatcher specifically designed for running workflows remotely on an Azure Virtual Machine. To configure the remote dispatcher, you will need to provide the following attributes:
+
+* `type: [str]`
+  * Currently we only support `Remote` as dispatcher type.
+* `config_path: [str]`
+  * The path to the dispatcher configuration file.
+
+Dispatcher configuration file is a json file that includes configurations for the dispatcher. As for Remote Dispatcher, you should include the following attributes:
+
+* `workflow_path: [str]`
+  * The path to run the Olive workflow on the remote dispatcher. The `cache_dir` and `output_dir` will be under this path. For example, the `olive_path` is `/home/ucm/olive`, and the `cache_dir` and `output_dir` in the config file are `cache` and `outputs`, then the cache will be saved in `/home/ucm/olive/cache` and the outputs will be saved in `/home/ucm/olive/outputs` on the remote dispatcher.
+* `conda_path: [str]`
+  * The path to conda to activate the environment in this session. Olive will run `source <conda_path>` to activate your conda. For example, the path could be `/opt/conda/etc/profile.d/conda.sh`.
+* `conda_name: [str]`
+  * The name of the conda environment to activate. Olive should be installed in this conda environment to run the workflow.
+* `hostname: [str]`
+  * The hostname of the remote dispatcher. Can be an IP address or a domain name.
+* `username: [str]`
+  * The username to login to the remote dispatcher.
+* `os: [str]`
+  * The operating system of the remote dispatcher. Can be "Linux" or "Windows".
+* `key_filename: [str]`
+  * The path to the private key file used to log in to the remote dispatcher. For example, the path could be `C:\Users\ucm\.ssh\key.pem` on Windows or `/home/ucm/.ssh/key.pem` on Linux.
+* `password: [str]`
+  * The password to login to the remote dispatcher. This is optional when `key_filename` is provided.
+
+Here is an example of remote dispatcher configuration file:
+
+```(json)
+{
+    "workflow_path": "/home/ucm/olive",
+    "conda_path": "/home/ucm/conda",
+    "conda_name": "olive",
+    "hostname": "8.8.8.8",
+    "os": "Linux",
+    "username": "ucm",
+    "key_filename": "/home/ucm/.ssh/key.pem"
+}
+```
+
+## Run Olive Workflow
+
+To run the Olive workflow, add a dispatcher configuation at the top level of the Olive config file:
```

Comment:
It might be simpler if we don't make the dispatcher a part of the `olive run` command and add a new command like `olive dispatch --config workflow.json --target remote_config.json`? `--retrieve` could be an option for this command.

Reply:
I've thought about this. The reason I didn't do this is because I would like to simplify the command to run Olive. But I'm open to this.